modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java

```diff
@@ -612,14 +613,19 @@ private void write(ClientMessagePacker packer, ChannelHandlerContext ctx) {
     }
 
     private void writeResponseHeader(
-            ClientMessagePacker packer, long requestId, ChannelHandlerContext ctx, boolean isNotification, boolean isError) {
+            ClientMessagePacker packer,
+            long requestId,
+            ChannelHandlerContext ctx,
+            boolean isNotification,
+            boolean isError,
+            long timestamp) {
         packer.packLong(requestId);
         writeFlags(packer, ctx, isNotification, isError);
 
         // Include server timestamp in error and notification responses as well:
         // an operation can modify data and then throw an exception (e.g. Compute task),
         // so we still need to update client-side timestamp to preserve causality guarantees.
-        packer.packLong(clockService.currentLong());
+        packer.packLong(Math.max(clockService.currentLong(), timestamp));
```

Comment:
It seems neither computing max nor overriding currentLong() is not necessary.
You can always send currentLong()
Can you elaborate why sending currentLong() is not enough ?